tools/run_tests/xds_k8s_test_driver/framework/infrastructure/gcp/compute.py

```diff
@@ -392,14 +393,56 @@ def wait_for_backends_healthy_status(
         *,
         timeout_sec: int = _WAIT_FOR_BACKEND_SEC,
         wait_sec: int = _WAIT_FOR_BACKEND_SLEEP_SEC,
-    ):
+    ) -> None:
+        if not backends:
+            raise ValueError("The list of backends to wait on is empty")
+
+        timeout = datetime.timedelta(seconds=timeout_sec)
         retryer = retryers.constant_retryer(
             wait_fixed=datetime.timedelta(seconds=wait_sec),
-            timeout=datetime.timedelta(seconds=timeout_sec),
+            timeout=timeout,
             check_result=lambda result: result,
         )
         pending = set(backends)
-        retryer(self._retry_backends_health, backend_service, pending)
+        try:
+            retryer(self._retry_backends_health, backend_service, pending)
+        except retryers.RetryError as retry_err:
+            unhealthy_backends: str
+            unhealthy_backends = ",".join([backend.name for backend in pending])
+
+            # Attempt to load backend health info for better debug info.
+            try:
+                unhealthy = []
+                for backend in pending:
+                    unhealthy.append(
+                        self.get_backend_service_backend_health(
```

Comment:
Just curious, this will only return unhealthy backend?

Reply:
No. At this moment everything is `pending` is not healthy, but it's possible that between we gave up and printed this error the backend transitioned to healthy. I'll add a comment there.